lib/tile.js

```diff
@@ -158,29 +142,33 @@ Tile.prototype.containsPoint = function(x,y) {
 }
 
 Tile.prototype.sameTypeAs = function(tile) {
-	return this.type == tile.type;
+	if(tile) { 
+		return this.type == tile.type;
+	} else {
+		return false;
+	}
 }
 
 Tile.prototype.possibleMatch = function() {
-	if(this.tileBelow() && this.tileBelow().sameTypeAs(this)) {
```

Comment:
Great refactor - I could see how it makes it slightly harder to read but I like how you only have to go to one method! Also possibly in the future use x,y instead of y,x - otherwise great!